backend/src/main/scala/cromwell/backend/standard/callcaching/StandardCacheHitCopyingActor.scala

```diff
@@ -0,0 +1,211 @@
+                                                   copiedDetritus: DetritusMap,
+                                                   returnCode: Option[Int]
+                                                  ) {
+    def remove(copyCommand: IoCopyCommand) = copy(copyCommandsToWaitFor = copyCommandsToWaitFor filterNot { _ == copyCommand })
+  }
+}
+
+class DefaultStandardCacheHitCopyingActor(standardParams: StandardCacheHitCopyingActorParams) extends StandardCacheHitCopyingActor(standardParams) with DefaultIoCommandBuilder
+
+/**
+  * Standard implementation of a BackendCacheHitCopyingActor.
+  */
+abstract class StandardCacheHitCopyingActor(val standardParams: StandardCacheHitCopyingActorParams)
+  extends FSM[StandardCacheHitCopyingActorState, Option[StandardCacheHitCopyingActorData]] with JobLogging with StandardCachingActorHelper with IoClientHelper { this: IoCommandBuilder =>
+
+  override lazy val jobDescriptor: BackendJobDescriptor = standardParams.jobDescriptor
+  override lazy val backendInitializationDataOption: Option[BackendInitializationData] = standardParams.backendInitializationDataOption
+  override lazy val serviceRegistryActor: ActorRef = standardParams.serviceRegistryActor
+  override lazy val configurationDescriptor: BackendConfigurationDescriptor = standardParams.configurationDescriptor
+  
+  lazy val destinationCallRootPath: Path = jobPaths.callRoot
+  lazy val destinationJobDetritusPaths: Map[String, Path] = jobPaths.detritusPaths
+  lazy val ioActor = standardParams.ioActor
+  
+  startWith(Idle, None)
+  
+  context.become(ioReceive orElse receive)
+  
+  /** Override this method if you want to provide an alternative way to duplicate files than copying them. */
+  protected def duplicate(copyPairs: Set[PathPair]): Option[Try[Unit]] = None
+  
+  when(Idle) {
+    case Event(CopyOutputsCommand(simpletons, jobDetritus, returnCode), None) =>
+      val sourceCallRootPath = lookupSourceCallRootPath(jobDetritus)
+
+      val processed = for {
+        (callOutputs, simpletonCopyPairs) <- processSimpletons(simpletons, sourceCallRootPath)
+        (destinationDetritus, detritusCopyPairs) <- processDetritus(jobDetritus)
+      } yield (callOutputs, destinationDetritus, simpletonCopyPairs ++ detritusCopyPairs)
+
+      processed match {
+        case Success((callOutputs, destinationDetritus, allCopyPairs)) =>
+          duplicate(allCopyPairs) match {
+            case Some(Success(_)) => succeedAndStop(returnCode, callOutputs, destinationDetritus)
+            case Some(Failure(failure)) => failAndStop(failure)
+            case None =>
+              // Should overwrite be true or false here ?
+              // We don't want to accidentally overwrite user's files
+              // OTOH what if cromwell is stopped in the middle of this and then restarted, it'll try to copy files again
+              // from the beginning, potentially overwriting the first ones.
```

Comment:
Can you explain this out a bit more long form? I see the basic idea but how would one wind up here, what are all the known implications, etc? Feel free to do so in person or slack if that's easier